internal/engine/filter/filter.go

```diff
@@ -67,6 +68,67 @@ func (f Filter) String() string {
 	return w.String()
 }
 
+func indent(query string, level int) string {
+	return strings.TrimRight(indentInternal(query, level), "\n ")
```

Comment:
Nit: to minimize conversions between `bytes.Buffer` and `string` I would suggest to create the buffer here and pass it to `indentInternal`. This could be reused in the recursive calls as well.